test/passing/tests/error4.ml.ref

```diff
@@ -6,6 +6,14 @@ File "tests/error4.ml", line 3, characters 8-16:
 3 | let b = (** ? *) ()          
             ^^^^^^^^
 Warning 50 [unexpected-docstring]: unattached documentation comment (ignored)
+File "tests/error4.ml", line 2, characters 0-13:
+2 | (** a or b *)
+    ^^^^^^^^^^^^^
+Warning 50 [unexpected-docstring]: ambiguous documentation comment
+File "tests/error4.ml", line 3, characters 8-16:
+3 | let b = (** ? *) ()          
+            ^^^^^^^^
+Warning 50 [unexpected-docstring]: unattached documentation comment (ignored)
```

Comment:
This happens because when warning 50 is reported, a valid parsetree is still returned. Look at the call to `Docstrings.warn_bad_docstrings` in the `wrap` function in `parsing/parse.ml` of the compiler. It seems like it would be enough to disable the warning during one of the calls to parse, probably better for the custom one. I'm not sure if the compiler's `Warnings.without_warnings` is exposed, but it ought to work I think.

Reply:
I've added a more specific argument `?disable_w50` to `Parse_with_comments.parse instead`, I previously tried to disable all warnings with `Warnings.without_warnings` but some warnings should not be disabled as it got me this diff:
```diff
------ test/failing/tests/unit_lex.ml.broken-ref
++++++ test/failing/unit_lex.ml.output
File "test/failing/tests/unit_lex.ml.broken-ref", line 1, characters 0-1:
-|File "tests/unit_lex.ml", line 18, characters 4-10:
-|18 |     ������ (* this file must be iso-8859-1 *)
-|         ^^^^^^
-|Alert deprecated: ISO-Latin1 characters in identifiers
 |ocamlformat: ignoring "tests/unit_lex.ml" (syntax error)
 |File "tests/unit_lex.ml", line 55, characters 2-8:
 |55 |   '\999'; (* wrong, but yet... *)
 |       ^^^^^^
 |Error: Illegal backslash escape in string or character ('\999'): 999 is outside the range of legal characters (0-255).
     patdiff (internal) (exit 1)
(cd _build/default && /home/gpe/.opam/ocamlformat-4.11.0/bin/patdiff -keep-whitespace -location-style omake test/failing/tests/unit_values.ml.broken-ref test/failing/unit_values.ml.output)
------ test/failing/tests/unit_values.ml.broken-ref
++++++ test/failing/unit_values.ml.output
File "test/failing/tests/unit_values.ml.broken-ref", line 1, characters 0-1:
-|File "tests/unit_values.ml", line 6, characters 10-11:
-|6 | let i32 = −1073741824, 1073741823
-|              ^
-|Alert deprecated: ISO-Latin1 characters in identifiers
 |ocamlformat: ignoring "tests/unit_values.ml" (syntax error)
 |File "tests/unit_values.ml", line 6, characters 11-12:
 |6 | let i32 = −1073741824, 1073741823
 |               ^
 |Error: Illegal character (\136)
     patdiff (internal) (exit 1)
(cd _build/default && /home/gpe/.opam/ocamlformat-4.11.0/bin/patdiff -keep-whitespace -location-style omake test/passing/tests/cinaps.ml test/passing/cinaps.ml.output)
------ test/passing/tests/cinaps.ml
++++++ test/passing/cinaps.ml.output
File "test/passing/tests/cinaps.ml", line 49, characters 0-1:
 |  (* fooooooo *)
 |  let y = 2 in
 |  (* foooooooo *)
 |  z
 |$*)
 |(*$*)
 |
 |let foo = foo
 |
 |(*$QR foo Q.small_int (fun i-> foo i (+) [1;2;3] = List.fold_left (+) i
 |  [1;2;3] ) *)
 |let foo = foo
 |
 |(* Cinaps comment should not wrap if they don't parse. The first one would
 |   crash and the second become a mess *)
 |
-|(*$(**)"
-|  "*)
+|(*$ (**) "\n  " *)
 |
 |(*$
 |  print_newline () ;
 |  <SYNTAX ERROR>
 |  List.iter
 |    (fun s -> Printf.printf "let ( %s ) = Pervasives.( %s )\n" s s)
 |    ["+"; "-"; "*"; "/"]
 |*)
 |(*$*)
```